### ISSUE TITLE
Cannot describe nested objects in request with RequestParam method. 

### ISSUE THREAD

**Issue by DCichosz** [FastEndpoints/FastEndpoints#151]
![image](https://user-images.githubusercontent.com/40369900/173250345-09965509-4e37-45a9-aaf7-1922eb61ed11.png)

**Comment by Arcalise08** [comment]
@dj-nitehawk I can tackle this one if nobody else is working on it.  It correlates fairly well with the work mentioned in #566 

Which is taking a little longer then expected but it's been one fire after another so far this year 😅

**Comment by dj-nitehawk** [comment]
yes please go ahead 🙏
and take your time. this one's pretty low priority.
cheers!

**Comment by sunecko** [comment]
Hello, I opened a pr I hope and it meets expectations. I remain on the lookout for any change request or whether it meets the requirements or not. I'm excited to finally be able to contribute to my favorite .NET library.

**Comment by dj-nitehawk** [comment]
@sunecko thank you for the contribution. but i think you've misunderstood the requirement here.
take the following example:
```cs
sealed class Request
{
    public int Id { get; set; }
    public NestedRequest Nested { get; set; }
}

sealed class NestedRequest
{
    public string Name { get; set; }
}

sealed class Summary : Summary<MyEndpoint, Request>
{
    public Summary()
    {
        RequestParam(r => r.Id, "this shows up in swagger doc");
        RequestParam(r => r.Nested.Name, "this should appear in swagger doc, but it doesn't");
    }
}

sealed class MyEndpoint : Endpoint<Request>
{
    public override void Configure()
    {
        Post("test");
        AllowAnonymous();
    }

    public override async Task HandleAsync(Request r, CancellationToken c) { }
}
```
which produces swagger schema like this:
![image](https://github.com/user-attachments/assets/df439512-fd66-43bb-8658-9eda075cc1cf)

the text description for the nested property is missing. that is the feature that's currently not supported. only the top level property descriptions can be provided like above atm.

**Comment by sunecko** [comment]
my mistake, not all the changes were added correctly to the commit, the main change is here
```c#
public class EndpointSummary<TRequest> : EndpointSummary where TRequest : notnull
{
    /// <summary>
    /// add a description for a request param for a given property of the request dto
    /// </summary>
    /// <param name="property">a member expression for specifying which property the description is for</param>
    /// <param name="description">the description text</param>
    public void RequestParam(Expression<Func<TRequest, object?>> property, string description)
    {
        var propertyPath = GetPropertyPath(property);
        Params[propertyPath] = description;
    }

    private string GetPropertyPath<TProperty>(Expression<Func<TRequest, TProperty>> expression)
    {
        if (expression.Body is MemberExpression member)
        {
            return member.ToString().Replace(expression.Parameters[0] + ".", "");
        }

        throw new ArgumentException("Invalid expression format", nameof(expression));
    }

    public new TRequest? ExampleRequest
    {
        get => (TRequest?)base.ExampleRequest;
        set => base.ExampleRequest = value;
    }
}
```

**Comment by dj-nitehawk** [comment]
looks alright...
but, see if you can use this existing extension method to get the dotted property path.
anyway pls push all changes to the PR branch so i can have a look at the diff.
cheers!

https://github.com/FastEndpoints/FastEndpoints/blob/db7a4c21ba6b82c0fb721517cfc70e635278d854/Src/Library/Validation/PropertyChainExtensions.cs#L8-L18

**Comment by sunecko** [comment]
change the solution I had given to use the extension method `GetPropertyChain()`

**Comment by sunecko** [comment]
![IMG-20241114-WA0005](https://github.com/user-attachments/assets/68974f46-8144-4146-8c61-3db0cc67ea78)

**Comment by dj-nitehawk** [comment]
@sunecko will review it soon. having a day off today :-)
btw, does it work for nested collection properties? `List<T>` etc?
for example if the property path is something like this?
```cs
r.SomeList[0].Name //with indexing
r.SomeProperty.SomeEnumerable.First().Name //indexing not possible due to IEnumerable<T>
```
see if you can crack that too ;-)

**Comment by sunecko** [comment]
ok, I'll check for the collections

**Comment by dj-nitehawk** [comment]
@sunecko 
i've merged #808 
thank you very much for your contribution.
